core/src/main/scala/com/microsoft/azure/synapse/ml/explainers/ICEFeature.scala

```diff
@@ -0,0 +1,168 @@
+      val name = fields("name") match {
+        case JsString(value) => value
+        case _ => throw new Exception("The name field must be a JsString.")
+      }
+      val numTopValues = fields.get("numTopValues") match {
+        case Some(JsNumber(value)) => Some(value.toInt)
+        case _ => None
+      }
+      val outputColName = fields.get("outputColName") match {
+        case Some(JsString(value)) => Some(value)
+        case _ => None
+      }
+      ICECategoricalFeature(name, numTopValues, outputColName)
+    }
+    override def write(obj: ICECategoricalFeature): JsValue = {
+      val map = Map("name" -> JsString(obj.name))++
+        obj.numTopValues.map("numTopValues" -> JsNumber(_))++
+        obj.outputColName.map("outputColName" -> JsString(_))
+      JsObject(map)
+    }
+  }
+  def fromMap(inputMap: java.util.HashMap[String, Any]): ICECategoricalFeature = {
+    val name: String = inputMap.get("name").toString
+    val numTopValues: Option[Int] = inputMap.get("numTopValues") match {
+      case value: Integer => Some(Integer2int(value))
+      case _ => None
+    }
+    val outputColName: Option[String] = inputMap.get("outputColName") match {
+      case value: String => Some(value)
+      case _ => None
+    }
+
+    ICECategoricalFeature(name, numTopValues, outputColName)
+  }
+}
+
+/**
+  * Represents a single numeric feature to be explained by ICE explainer.
+  * @param name The name of the numeric feature.
+  * @param numSplits The number of splits for the value range for the numeric feature.
+  *                  Default: 10.0
+  * @param rangeMin Specifies the min value of the range for the numeric feature. If not specified,
+  *                 it will be computed from the background dataset.
+  * @param rangeMax Specifies the max value of the range for the numeric feature. If not specified,
+  *                 it will be computed from the background dataset.
+  * @param outputColName The name for output column with explanations for the feature.
+  *                      Default: input name of the feature + "_dependence"
+  */
+case class ICENumericFeature(override val name: String, numSplits: Option[Int] = None,
+                             rangeMin: Option[Double] = None, rangeMax: Option[Double] = None,
+                             outputColName: Option[String] = None)
+  extends ICEFeature(name, outputColName) {
+  override def validate: Boolean = {
+    // rangeMax and rangeMin may not be specified, but if specified: rangeMin <= rangeMax.
+    numSplits.forall(_ > 0) && (rangeMax.isEmpty || rangeMin.isEmpty || rangeMin.get <= rangeMax.get)
+  }
+
+  private val defaultNumSplits = 10
+  def getNumSplits: Int = {
+    this.numSplits.getOrElse(defaultNumSplits)
+  }
+}
+
+/**
+  * Companion object to provide JSON serializer and deserializer for ICENumericFeature.
+  */
+object ICENumericFeature {
+  implicit val JsonFormat: JsonFormat[ICENumericFeature] = new JsonFormat[ICENumericFeature] {
```

Comment:
likewise here, seems like this is alot of work for json parsing and the default impl for case classes should do the trick